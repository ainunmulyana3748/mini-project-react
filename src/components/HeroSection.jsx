import React from "react";
import Button from "./Button";

const HeroSection = () => {
  return (
    <div className="relative h-screen w-full">
      <div className="absolute inset-0 bg-black/50 z-0" />
      <div
        className="absolute inset-0 bg-cover bg-center z-[-1]"
        style={{
          backgroundImage:
            "url('https://ik.imagekit.io/tvlk/image/imageResource/2024/06/12/1718188327775-eea839531a5ec5b6b7a784c4e61d1ceb.png?tr=q-75')",
          backgroundSize: "cover",
          backgroundPosition: "center",
          backgroundRepeat: "no-repeat",
        }}
      />
      <div className="relative z-10 flex flex-col items-center justify-center h-full text-center text-white px-4 max-w-screen-xl mx-auto">
        <h1 className="text-4xl md:text-5xl font-bold mb-4 drop-shadow-md">
          Ditemenin.
        </h1>
        <p className="text-base md:text-lg max-w-xl mb-6 drop-shadow-sm">
          Kemanapun dan kapanpun, mau jalan-jalan dan butuh pendamping? Yuk,
          klik tombol di bawah ini dan mulai petualanganmu sekarang!
        </p>
        <Button variant="primary">Lanjutkan</Button>
      </div>
    </div>
  );
};

export default HeroSection;
