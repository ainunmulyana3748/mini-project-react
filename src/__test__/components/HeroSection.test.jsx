import { render, screen, fireEvent } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";

// Mock useNavigate
const mockedUsedNavigate = jest.fn();
jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useNavigate: () => mockedUsedNavigate,
}));

import HeroSection from "../../components/HeroSection";

describe("HeroSection Component", () => {
  test("render judul, deskripsi, dan tombol", () => {
    render(
      <BrowserRouter>
        <HeroSection />
      </BrowserRouter>
    );
    expect(screen.getByText("Ditemenin.")).toBeInTheDocument();
    expect(
      screen.getByText(/Kemanapun dan kapanpun, mau jalan-jalan/)
    ).toBeInTheDocument();
    expect(screen.getByText("Lanjutkan")).toBeInTheDocument();
  });

  test("klik tombol Lanjutkan memanggil navigate", () => {
    render(
      <BrowserRouter>
        <HeroSection />
      </BrowserRouter>
    );
    fireEvent.click(screen.getByText("Lanjutkan"));
    expect(mockedUsedNavigate).toHaveBeenCalledWith("/");
  });

  test("snapshot hero section", () => {
    const { container } = render(
      <BrowserRouter>
        <HeroSection />
      </BrowserRouter>
    );
    expect(container).toMatchSnapshot();
  });
});
