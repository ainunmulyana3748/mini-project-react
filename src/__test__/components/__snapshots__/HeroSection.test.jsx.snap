// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeroSection Component snapshot hero section 1`] = `
<div>
  <div
    class="relative h-screen w-full"
  >
    <div
      class="absolute inset-0 bg-black/50 z-0"
    />
    <div
      class="absolute inset-0 bg-cover bg-center z-[-1]"
      style="background-image: url("https://ik.imagekit.io/tvlk/image/imageResource/2024/06/12/1718188327775-eea839531a5ec5b6b7a784c4e61d1ceb.png?tr=q-75"); background-size: cover; background-position: center; background-repeat: no-repeat;"
    />
    <div
      class="relative z-10 flex flex-col items-center justify-center h-full text-center text-white px-4 max-w-screen-xl mx-auto"
    >
      <h1
        class="text-4xl md:text-5xl font-bold mb-4 drop-shadow-md"
      >
        Ditemenin.
      </h1>
      <p
        class="text-base md:text-lg max-w-xl mb-6 drop-shadow-sm"
      >
        Kemanapun dan kapanpun, mau jalan-jalan dan butuh pendamping? Yuk, klik tombol di bawah ini dan mulai petualanganmu sekarang!
      </p>
      <button
        class="px-4 py-2 rounded font-medium bg-black text-white hover:bg-zinc-900 focus:ring-black disabled:opacity-50 cursor-pointer"
      >
        Lanjutkan
      </button>
    </div>
  </div>
</div>
`;
